weave

```diff
@@ -1037,6 +1042,10 @@ launch_router() {
             exit 1
         fi
     fi
+    COVERAGE_ARGS=""
+    if [ -n "$COVERAGE" ] ; then
+        COVERAGE_ARGS="-test.coverprofile=/home/weave/cover.router.prof --"
+    fi
```

Comment:
I'd put this code in the top-level, and then in `ask_version` reset `COVERAGE_ARGS=` if the container is not the router. Removes the duplication.